### PR TITLE
[DependencyInjection] Add service_closure() to the PHP-DSL

### PR DESCRIPTION
**Pull request by HypeMC**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/15730 and https://github.com/symfony/symfony-docs/pull/15731

Adds a `service_closure()` function to the PHP-DSL.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/config/services_closure_argument.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+return function (ContainerConfigurator $c) {
+    $s = $c->services()->defaults()->public();
+
+    $s->set('foo', 'Foo');
+
+    $s->set('service_closure', 'Bar')
+        ->args([service_closure(service('foo'))]);
+
+    $s->set('service_closure_invalid', 'Bar')
+        ->args([service_closure(service('invalid_service')->nullOnInvalid())]);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/config/services_closure_argument.php]: it'd be great to be able to do this instead:
`service_closure('invalid_service')->nullOnInvalid()`

maybe with a new `ClosureReferenceConfigurator`?

Comment (HypeMC) [reply src/Symfony/Component/DependencyInjection/Tests/Fixtures/config/services_closure_argument.php]: @nicolas-grekas Done. The fabbot failure is unrelated to my changes, I can apply it if you want.

**Review by nicolas-grekas — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Comment by fabpot** [comment]
@HypeMC Can you submit a PR on the docs? Or at least, give a full usage example in the PR description to help the doc team? Thank you.

**Comment by HypeMC** [comment]
@fabpot Sure, I'll submit a PR sometime tomorrow.

**Comment by HypeMC** [comment]
@fabpot PR(s) on the docs  added: https://github.com/symfony/symfony-docs/pull/15730 and https://github.com/symfony/symfony-docs/pull/15731

**Comment by nicolas-grekas** [comment]
Thank you @HypeMC.

**Pull request merged**